internal/testrand/rand.go

```diff
@@ -0,0 +1,89 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package testrand provides random generation and flags for testing.
+package testrand
+
+import (
+	"flag"
+
+	"golang.org/x/exp/rand"
+)
+
+var (
+	seedFlag    = flag.Uint64("testrand.seed", 1, "random seed for tests (0=randomize)")
+	extremeFlag = flag.Float64("testrand.extreme", 0, "probability of returning extreme values")
+	nanFlag     = flag.Float64("testrand.nan", 0, "probability of returning nan")
+)
+
+// TB is an interfaces that corresponds to *testing.T and *testing.B
+type TB interface {
+	Logf(format string, args ...interface{})
+}
+
+// Source corresponds to the interface in golang.org/x/exp/rand.Source.
+type Source interface {
+	Uint64() uint64
+	Seed(seed uint64)
+}
+
+// Rand corresponds to golang.org/x/exp/rand.Rand.
+type Rand interface {
```

Comment:
Same here.

Reply:
Here we need the interface since `rand.Rand` is a concrete struct and to support the extreme value generation we need a different implementation.

Reply:
Sorry, you're right.